lib/slack/handlers.ex

```diff
@@ -48,6 +48,14 @@ defmodule Slack.Handlers do
     end
   end)
 
+  def handle_slack(%{type: "im_created", channel: channel, user: user}, slack) do
+    {:ok, put_in(slack, [:ims, channel.id], %{id: channel.id, user: user})}
+  end
+
+  def handle_slack(%{type: "im_close", channel: channel}, slack) do
+    {:ok, put_in(slack, [:ims], Map.delete(slack[:ims], channel))}
+  end
```

Comment:
@BlakeWilliams Added im create/close events here, these are fired after the [im.open](https://api.slack.com/methods/im.open) and [im.close](https://api.slack.com/methods/im.close) methods are called.

To make it easier on the user, I've added a `Slack.Web` module which has the beginnings of the Web API for areas like this where websocket support is lacking.